lib/game.rb

```diff
@@ -0,0 +1,144 @@
+  
+  def initialize(player_1=Players::Human.new('X'), player_2=Players::Human.new('O'), board=Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board 
+  end
+  
+  def self.start
+    play = true 
+    while play
+      puts "How many players?"
+      player_num = choose_player 
+      if player_num == "0"
+        game = Game.new(player_1=Players::Computer.new('X'), player_2=Players::Computer.new('O'))
+      elsif player_num == "1"
+        puts "Who will go first? Press c for the computer or h for human."
+        player1 = gets.chomp 
+          if player1 == 'c'
+            game = Game.new(player_1=Players::Computer.new('X'), player_2=Players::Human.new('O'))
+          else 
+            game = Game.new(player_1=Players::Human.new('X'), player_2=Players::Computer.new('O'))
+          end
+      else 
+        game = Game.new 
+      end
+      puts "Let's play!"
+      game.play 
+      puts "Nice job! Would you like to play again? y/n"
+      input = gets.chomp 
+      input == 'y' ? play : play = false 
+    end
+    puts "Buh bye!"
+  end
+  
+  def self.war
+    war_quotes = ["Cry 'Havoc!', and let slip the dogs of war. -Shakespeare, Julius Caesar", "The two most powerful warriors are patience and time. -Leo Tolstoy", "You must not fight too often with one enemy, or you will teach him all your art of war. -Napolean Bonaparte", "The supreme art of war is to subdue the enemy without fighting. -Sun Tzu", "To be prepared for war is one of the most effective ways of preserving peace. -George Washington"]
+    war_log = {x: 0, o: 0, cat: 0}
+    100.times do
+      game = Game.new(player_1=Players::Computer.new('X'), player_2=Players::Computer.new('O'))
+      until game.over?
+        game.turn
+      end
+      if game.won?
+        war_log[winner.downcase.to_sym] += 1
+      else
+        war_log[:cat] += 1
+      end
+    end
+    puts war_quotes.sample
+    puts war_log
+  end
+  
+  # #Game.start helper methods 
+  def self.choose_player
+    valid_choices = ["0", "1", "2", "wargames"]
+    player_num = "-1"
+    puts "How many players?"
+    until valid_choices.include?(player_num)
+      puts "Please enter 1, for one-player mode, 2 for two-player mode,"
+      puts "or 0 for the computer to play itself."
+      player_num = gets.chomp
+    end
+    player_num
+  end
+      
+  # returns currently player based on number of turns already taken 
+  # board object tracks turns already taken 
+  def current_player
+    board.turn_count.even? ? player_1 : player_2
+  end
+
+  # Validation methods:
+  
+  # checks WIN_COMBINATIONS and board too look for a winner, returns winning combination
+  def won?
+    winner = nil
+    WIN_COMBINATIONS.each do |combo|
+      converted_combo = combo.collect{ |i| board.cells[i] }
+      if converted_combo.all?{|cell| cell == "X"} || converted_combo.all?{|cell| cell == "O"}
+        winner = combo 
+      end
+    end
+    winner 
+  end
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```